### PR TITLE
[Fixed #26191] Need To Change Label In Totals Block in Order View Page

### LINKED ISSUE #26191: Need To Change Label In Totals Block in Order View Page
After Place, the order, Create an Invoice Of some items. After that Cancel, the order, Total of Cancel items display as Total Due. 

### Preconditions (*)

Magento 2.3.3

### Steps to reproduce (*)
1. Place Order with 5 Items.
2. Now Create Invoice For 1st 3 Items.
3. Cancel Order.

### Expected result (*)
1. If Order Cancel, then Change the label of Total Due.

### Actual result (*)
1. If Order Canceled, Then need to Change the label  "Total Cancel" instead of "Total Due".

### PR DESCRIPTION
**Pull request by madhurajawat**
### Description (*)

Need To Change Label In Totals Block in Order View Page # 26191

### Fixed Issues (if relevant)

1. magento/magento2#26191: Need To Change Label In Totals Block in Order View Page

### Manual testing scenarios (*)

  1. Place Order with 5 Items.
   2. Create Invoice For 1st 3 Items.
   3.  Cancel Order.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by lbajsarowicz** [comment]
Could you provide more descriptive title to your PR? For future reference, such as searching that is really helpful :-)

**Comment by madhu-ranosys** [comment]
> Could you provide more descriptive title to your PR? For future reference, such as searching that is really helpful :-)

Sure, Updated PR title.

**Comment by gwharton** [comment]
Total cancel is not great english. Should it be total canceled. Also translatable strings should be added to module i18n en_us.csv file.

**Comment by madhu-ranosys** [comment]
> Total cancel is not great english. Should it be total canceled. Also translatable strings should be added to module i18n en_us.csv file.

Yes, We can update it with total canceled and translatable strings.

**Comment by madhu-ranosys** [comment]
> Total cancel is not great english. Should it be total canceled. Also translatable strings should be added to module i18n en_us.csv file.

Submitted PR with the changes.

**Comment by madhu-ranosys** [comment]
> > Total cancel is not great english. Should it be total canceled. Also translatable strings should be added to module i18n en_us.csv file.
> 
@gwharton Submitted PR with the changes.

**Comment by swnsma** [comment]
Hi @madhu-ranosys !
Thank you for your contribution.
Could you please sign Adobe CLA? https://github.com/magento/magento2/pull/27516/checks?check_run_id=586465685
Thank you!

**Review by swnsma — CHANGES_REQUESTED** [review]
Hi @madhu-ranosys !
Thank you for your contribution.
Could you please sign Adobe CLA? https://github.com/magento/magento2/pull/27516/checks?check_run_id=586465685
Thank you!

**Comment by madhurajawat** [comment]
> Hi @madhu-ranosys !
> Thank you for your contribution.
> Could you please sign Adobe CLA? https://github.com/magento/magento2/pull/27516/checks?check_run_id=586465685
> Thank you!

It's done from my side.

**Comment by madhurajawat** [comment]
@swnsma please help to verify.

**Comment by swnsma** [comment]
Hi @madhurajawat,
I see that you have set this PR as a 'Draft'.
Please set is as 'ready for review' as soon as you will realize that this is ready for review.
Also, please check failed static tests.

**Comment by madhurajawat** [comment]
> Hi @madhurajawat,
> I see that you have set this PR as a 'Draft'.
> Please set is as 'ready for review' as soon as you will realize that this is ready for review.
> Also, please check failed static tests.

I have set  'ready for review'. 
Thanks.

**Comment by madhurajawat** [comment]
@swnsma Can you please help me to check failed static tests This is my first contribution So can you please let me know or gyidence how can I check it further or give some specific error if any
Thanks.

**Review by swnsma — CHANGES_REQUESTED** [review]
Hello @madhurajawat,

Yes, sure.
Screen Shot 2020-04-22 at 11 04 33 PM

Please click for "Details" near filed tests.
You will be forwarded to check details.
Screen Shot 2020-04-22 at 11 04 48 PM
Here you can select a build and check it's report.
Screen Shot 2020-04-22 at 11 05 03 PM

There you can see amount of failed tests.
And there you can see the test output.
Screen Shot 2020-04-22 at 11 05 31 PM

**Comment by madhurajawat** [comment]
@swnsma Fixed static test issue. Please help to verify.

**Comment by VladimirZaets** [comment]
@magento run all tests

**Comment by madhurajawat** [comment]
@swnsma Please help to verify further.

**Comment by swnsma** [comment]
@magento run all tests

File: app/code/Magento/Sales/Model/Order.php [hunk 1]
```diff
@@ -1816,6 +1816,30 @@ public function getTotalDue()
         $total = $this->priceCurrency->round($total);
         return max($total, 0);
     }
+    
+    /**
+     * Retrieve order total due or cancel label
+     *
+     * @return float|null
+     */
+    public function getTotalDueCancelLabel()
```
Comment (swnsma) [inline app/code/Magento/Sales/Model/Order.php]: I do not think that `Order` should be directly responsible for the text of the label.
Could we try to keep this logic isolated in totals block (app/code/Magento/Sales/Block/Adminhtml/Order/Totals.php)?

You can use as an example: https://github.com/nit-it/magento2/pull/2/files

**Review by swnsma — CHANGES_REQUESTED** [review]
Hi @madhurajawat!
Sorry for delay.
Could you check my comment?

File: app/code/Magento/Sales/Model/Order.php (hunk 1 above)
Comment (madhurajawat) [reply app/code/Magento/Sales/Model/Order.php]: @swnsma Please help to verify further.

**Comment by madhurajawat** [comment]
@swnsma Can you please help to check this further

**Comment by swnsma** [comment]
@magento run all tests

**Comment by madhurajawat** [comment]
@swnsma Can you please help to check this further

**Comment by swnsma** [comment]
Hi @madhurajawat!
Sorry for waiting, finally - tests are green.
According to Magento Definition of Done I need to ask you to write tests which could test new behaviour (changing of the label). 
Will you proceed with tests?

**Comment by madhurajawat** [comment]
>  write tests which could test new behaviour (changing of the label).

**Comment by madhurajawat** [comment]
> Hi @madhurajawat!
> Sorry for waiting, finally - tests are green, which means that everything OK with the code.
> According to Magento Definition of Done I need to ask you to write tests which could test new behaviour (changing of the label).
> Will you proceed with tests?

Sorry It was by mistake closed.  As i Have no idea about the  write tests Could you please  guide me for the same. what to do further?

**Comment by swnsma** [comment]
Hi @madhurajawat 
As your changes is mostly related to Frontend, better is to add new checks with Functional Tests.
You can read more about Magento Functional Tests Framework here: https://devdocs.magento.com/mftf/docs/introduction.html

As example of tests, you can check the folder: app/code/Magento/Sales/Test/Mftf/Test.
In the article(s) you can find the information how to configure and run tests.

**Review by swnsma — CHANGES_REQUESTED** [review]
Please cover changes with functional (MFTF) tests.

**Comment by madhurajawat** [comment]
> Please cover changes with functional (MFTF) tests.

@swnsma Sure.

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
I will take care of test coverage

**Comment by engcom-Echo** [comment]
@magento run all tests

**Review by swnsma — APPROVED** [review]

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed
Was checked case from [Manual testing scenarios](https://github.com/magento/magento2/pull/27516#issue-396357038)
**Before:**
#27516Main

**After:**
#27516PR

**Pull request merged**